qrcode_detection_package/main.py

```diff
@@ -1,56 +1,358 @@
-import sys
 import rclpy
+import time
+import subprocess
 from rclpy.node import Node
 import cv2
 from std_msgs.msg import String
 
 from common_package_py.common_node import CommonNode
+from interfaces.msg import QRCodeInfo
+from interfaces.msg import Control
 
 
 class QRCodeScannerNode(CommonNode):
     def __init__(self, id: str):
+        """
+        Constructor of the QRCodeScannerNode class.
+
+        This method initializes the object properties for the QRCodeSearch object.
+        It sets the initial state to "ready" and configures the subscription and publisher objects for control 
+        and QR code information. Additionally, it initializes the QR code detector from OpenCV.
+        The method also includes a configuration variable for selecting the QR code detection method.
+        This method is called automatically when creating a QRCodeSearch object.
+
+        Args: 
+            id: The node_id.
+
+        Returns: 
+            None
+        """
         super().__init__(id)
+        self.set_state("ready")
+        self.control_subscription = self.create_subscription(
+            Control,
+            "control",
+            self.__callback_control,
+            10
+        )
         self.qr_code_detector = cv2.QRCodeDetector()
         self.qr_publisher = self.create_publisher(
-            String, 'qr_codes', 10)
+            QRCodeInfo, 'qr_code_info', 10)
+
+        # until tested properly different detection methods are implemented and can be configured here
+        # 0 = load stored test image (for testing purposes)
+        # 1 = use OpenCV to get image from camera (this is the wanted solution)
+        # 2 = use libcamera shell script to take photo and load it (fallback solution)
+        self.config_detection_method = 0
+
+    def __callback_control(self, control_msg):
+        """
+        Callback function for receiving control messages.
+
+        This callback function is triggered when a control message is received.
+        It checks if the target ID of the control message matches the QR code scanner node.
+        If it does, the function logs a message indicating the receipt of the control message.
+        Depending on the 'active' flag in the control message, the function either activates
+        or deactivates the QR code scanner node by calling the respective private methods.
+
+        Args: 
+            control_msg: The received control message.
+
+        Returns: 
+            None
+        """
+        if (control_msg.target_id == "qr_code_scanner_node"):
+            self.get_logger().info("Recieved control message")
+            if (control_msg.active):
+                self._activate_()
+            else:
+                self._deactivate_()
+
+    def set_state(self, state):
```

Comment:
If you already have a function for it, you could also add a debug or info message here